### PR TITLE
Add support for S3 notifications to EventBridge

### PR DESCRIPTION
**Pull request by pinzon**
This PR addresses issue #5994. With these changes, the S3 Notifications can be send to the EventBridgeService.

### CODE REVIEW

File: localstack/services/s3/s3_listener.py [hunk 1]
```diff
@@ -380,12 +389,58 @@ def send_notification_for_subscriber(
...
+            )
+
+        entry["Detail"] = json.dumps(entry["Detail"])
+
+        try:
+            events_client.put_events(Entries=[entry])
+        except Exception as e:
+            LOGGER.warning(
+                f'Unable to send notification for S3 bucket "{bucket_name}" to EventBridge', e
+            )
```
Comment (thrau) [inline localstack/services/s3/s3_listener.py:441]: maybe we could log this using the exception level

```suggestion
            LOGGER.exception(
                'Unable to send notification for S3 bucket "%s" to EventBridge', bucket_name
            )
```

**Review by thrau — APPROVED** [review]
LGTM @pinzon. would be great if we could think about improving the parity checks here with snapshot tests. maybe @dominikschubert can give an intro at some point!

**Comment by pinzon** [comment]
@thrau now this test uses snapshots :grin:

**Review by thrau — APPROVED** [review]
very cool! thanks @pinzon

**Pull request merged**